### ISSUE TITLE
lsregistration service not working on 5.0.6 version Ubuntu 20.04

### ISSUE THREAD

**Issue by Rezi1999** [perfsonar/ls-registration-daemon#66]
after updating Perfsonar from 5.0.5 to 5.0.6 lsregistration service not working 

```
root@ps:~# systemctl status perfsonar-lsregistrationdaemon.service 
● perfsonar-lsregistrationdaemon.service - perfSONAR Lookup Service Registration Daemon
     Loaded: loaded (/lib/systemd/system/perfsonar-lsregistrationdaemon.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Wed 2023-11-15 13:15:05 +04; 3min 34s ago
    Process: 123147 ExecStart=/usr/lib/perfsonar/bin/lsregistrationdaemon.pl --config=/etc/perfsonar/lsregistrationdaemon.conf --logger=/etc/perfsonar/lsregistrationdaemon-logger.conf --user=perfsonar --group=p>

Nov 15 13:15:04 ps systemd[1]: Starting perfSONAR Lookup Service Registration Daemon...
Nov 15 13:15:05 ps lsregistrationdaemon.pl[123147]: Config::General The file "/etc/perfsonar/lsregistrationdaemon.conf" does not exist within ConfigPath: /etc/perfsonar! at /usr/lib/perfsonar/bin/lsregistration>
Nov 15 13:15:05 ps systemd[1]: perfsonar-lsregistrationdaemon.service: Control process exited, code=exited, status=2/INVALIDARGUMENT
Nov 15 13:15:05 ps systemd[1]: perfsonar-lsregistrationdaemon.service: Failed with result 'exit-code'.
Nov 15 13:15:05 ps systemd[1]: Failed to start perfSONAR Lookup Service Registration Daemon
```

It says config file does not exist:

`General The file "/etc/perfsonar/lsregistrationdaemon.conf" does not exist within ConfigPath: /etc/perfsonar! at /usr/lib/perfsonar/bin/lsregistration`

**Comment by laeti-tia** [comment]
We already had that reported in https://github.com/perfsonar/toolkit/issues/447 and then closed it because it seemed to be working.  It would be good to have a better understanding of what is going on.

**Comment by arlake228** [comment]
I think this is the problem: https://github.com/perfsonar/ls-registration-daemon/blob/master/lsregistrationdaemon/perfsonar-lsregistrationdaemon/unibuild-packaging/deb/perfsonar-lsregistrationdaemon.maintscript. Can we remove this? I don't think it is needed. 

I have also encountered this a number of times doing updates on Ubuntu.

**Comment by igarny** [comment]
Hello 
I am not able to confirm this on my install with an upgrade from 5.0.4:
```
~$ tail -n 20 /var/log/perfsonar/lsregistrationdaemon.log
2023/12/07 09:19:21 (3324349) INFO> Base.pm:358 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_refresh - Record is up, adding registration to refresh list(key=lookup/service/06f40f9a-48bd-4fdc-aba6-0d2ebe2fb41c, description=GNT-MP-u Frankfurt OWAMP Server)
2023/12/07 09:19:21 (3324349) INFO> Base.pm:358 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_refresh - Record is up, adding registration to refresh list(key=lookup/service/3f126a5f-35fc-43ae-bd43-b06b5de4d9c8, description=GNT-MP-u Frankfurt TWAMP Server)
2023/12/07 09:19:21 (3324349) INFO> Base.pm:353 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_refresh - Record is up, registering (description=GNT-MP-u Frankfurt pScheduler)
2023/12/07 09:19:21 (3324349) WARN> lsregistrationdaemon.pl:143 main::__ANON__ - Warned: Use of uninitialized value in numeric eq (==) at /usr/lib/perfsonar/bin/../lib/perfSONAR_PS/LSRegistrationDaemon/Base.pm line 521.
2023/12/07 09:19:22 (3324349) INFO> Base.pm:537 perfSONAR_PS::LSRegistrationDaemon::Base::register - Service registered. Next Refresh: 1702023562(key=lookup/service/2f300122-8911-4efd-81cb-284bcddf7826, description=GNT-MP-u Frankfurt pScheduler)
2023/12/07 09:19:22 (3324349) INFO> Base.pm:377 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_refresh - Calling bulk_keepalive()
2023/12/07 09:19:22 (3324349) INFO> Base.pm:419 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_keepalive - Bulk keepalive succeeded.
2023/12/07 09:19:22 (3324349) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/host/eaa79780-54c0-4611-b91f-cc10a7b7b765with refresh1701944662
2023/12/07 09:19:22 (3324349) WARN> lsregistrationdaemon.pl:143 main::__ANON__ - Warned: Use of uninitialized value in concatenation (.) or string at /usr/lib/perfsonar/bin/../lib/perfSONAR_PS/LSRegistrationDaemon/Host.pm line 366.
2023/12/07 09:19:22 (3324349) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701944662(key=lookup/host/eaa79780-54c0-4611-b91f-cc10a7b7b765, description=)
2023/12/07 09:19:22 (3324349) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/interface/14f45db8-a016-42d6-89ac-50e729b681f2with refresh1701944662
2023/12/07 09:19:22 (3324349) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701944662(key=lookup/interface/14f45db8-a016-42d6-89ac-50e729b681f2, description=eth0)
2023/12/07 09:19:22 (3324349) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/interface/19761a69-f005-4e2f-957f-ea30816fff8bwith refresh1701944662
2023/12/07 09:19:22 (3324349) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701944662(key=lookup/interface/19761a69-f005-4e2f-957f-ea30816fff8b, description=eth2)
2023/12/07 09:19:22 (3324349) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/service/06f40f9a-48bd-4fdc-aba6-0d2ebe2fb41cwith refresh1701944662
2023/12/07 09:19:22 (3324349) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701944662(key=lookup/service/06f40f9a-48bd-4fdc-aba6-0d2ebe2fb41c, description=GNT-MP-u Frankfurt OWAMP Server)
2023/12/07 09:19:22 (3324349) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/interface/1b117cde-3db9-4112-8d79-81665da06d4fwith refresh1701944662
2023/12/07 09:19:22 (3324349) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701944662(key=lookup/interface/1b117cde-3db9-4112-8d79-81665da06d4f, description=eth1)
2023/12/07 09:19:22 (3324349) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/service/3f126a5f-35fc-43ae-bd43-b06b5de4d9c8with refresh1701944662
2023/12/07 09:19:22 (3324349) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701944662(key=lookup/service/3f126a5f-35fc-43ae-bd43-b06b5de4d9c8, description=GNT-MP-u Frankfurt TWAMP Server)
```

**Comment by igarny** [comment]
Hello guys
I am not able to confirm this on a clean install of pS 5.0.6
```
psmp-uat-mgmt-fra-de:~$ tail -n 20 /var/log/perfsonar/lsregistrationdaemon.log
2023/12/07 10:13:50 (2413649) INFO> Base.pm:358 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_refresh - Record is up, adding registration to refresh list(key=lookup/service/827371cd-4631-4f07-87b9-d8c0f5a01a20, description=GNT-MP UAT Frankfurt TWAMP Server)
2023/12/07 10:13:50 (2413649) INFO> Base.pm:353 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_refresh - Record is up, registering (description=GNT-MP UAT Frankfurt pScheduler)
2023/12/07 10:13:50 (2413649) WARN> lsregistrationdaemon.pl:143 main::__ANON__ - Warned: Use of uninitialized value in numeric eq (==) at /usr/lib/perfsonar/bin/../lib/perfSONAR_PS/LSRegistrationDaemon/Base.pm line 521.
2023/12/07 10:13:50 (2413649) ERROR> Base.pm:540 perfSONAR_PS::LSRegistrationDaemon::Base::register - Problem registering service. Will retry full registration next time: HTTP/1.1 403 Forbidden
(key=NONE, description=GNT-MP UAT Frankfurt pScheduler)
2023/12/07 10:13:50 (2413649) INFO> Base.pm:377 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_refresh - Calling bulk_keepalive()
2023/12/07 10:13:51 (2413649) INFO> Base.pm:419 perfSONAR_PS::LSRegistrationDaemon::Base::bulk_keepalive - Bulk keepalive succeeded.
2023/12/07 10:13:51 (2413649) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/interface/2fa9dfb6-196e-4363-b216-f282829b3799with refresh1701947931
2023/12/07 10:13:51 (2413649) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701947931(key=lookup/interface/2fa9dfb6-196e-4363-b216-f282829b3799, description=ens1f0)
2023/12/07 10:13:51 (2413649) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/service/827371cd-4631-4f07-87b9-d8c0f5a01a20with refresh1701947931
2023/12/07 10:13:51 (2413649) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701947931(key=lookup/service/827371cd-4631-4f07-87b9-d8c0f5a01a20, description=GNT-MP UAT Frankfurt TWAMP Server)
2023/12/07 10:13:51 (2413649) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/interface/c29022d4-b790-417c-9402-068af961c3ebwith refresh1701947931
2023/12/07 10:13:51 (2413649) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701947931(key=lookup/interface/c29022d4-b790-417c-9402-068af961c3eb, description=eno1)
2023/12/07 10:13:51 (2413649) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/service/933409f1-9dc6-4945-87c0-62cb82930e8fwith refresh1701947931
2023/12/07 10:13:51 (2413649) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701947931(key=lookup/service/933409f1-9dc6-4945-87c0-62cb82930e8f, description=GNT-MP UAT Frankfurt OWAMP Server)
2023/12/07 10:13:51 (2413649) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/host/0b466e44-5b2f-447a-9655-5260fd612d2cwith refresh1701947931
2023/12/07 10:13:51 (2413649) WARN> lsregistrationdaemon.pl:143 main::__ANON__ - Warned: Use of uninitialized value in concatenation (.) or string at /usr/lib/perfsonar/bin/../lib/perfSONAR_PS/LSRegistrationDaemon/Host.pm line 366.
2023/12/07 10:13:51 (2413649) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701947931(key=lookup/host/0b466e44-5b2f-447a-9655-5260fd612d2c, description=)
2023/12/07 10:13:51 (2413649) INFO> Base.pm:666 perfSONAR_PS::LSRegistrationDaemon::Base::update_key - Updated key: lookup/interface/aa22b065-4b45-49f3-9c9c-2264133d1534with refresh1701947931
2023/12/07 10:13:51 (2413649) INFO> Base.pm:483 perfSONAR_PS::LSRegistrationDaemon::Base::_handle_bulk_update_success - Service renewed. Next Refresh: 1701947931(key=lookup/interface/aa22b065-4b45-49f3-9c9c-2264133d1534, description=eno2)

**Comment by laeti-tia** [comment]
> I think this is the problem: https://github.com/perfsonar/ls-registration-daemon/blob/master/lsregistrationdaemon/perfsonar-lsregistrationdaemon/unibuild-packaging/deb/perfsonar-lsregistrationdaemon.maintscript. Can we remove this? I don't think it is needed.

This is part of the problem.  In fact there are also the following commits involved:
- 6cad028029d20e2e43115d64c4536625378c4eb5 which solved #32 
- f49752a5e458d4b754f62f1fa291b93a369d36a4 which solved #65 but dropped the solution to #32

For now I did remove the `rm_conffile` maintscript and this keeps the `lsregistrationdaemon.conf` in place even if it has been modified.  But, since f49752a5e458d4b754f62f1fa291b93a369d36a4 and the 5.0 release, we're back to the situation before #32 where auto-update will fail if we ship a new version of the default `lsregistrationdaemon.conf` in an update.

We could do a thing similar to what we did in 6cad028029d20e2e43115d64c4536625378c4eb5 but we need to place the original `lsregistrationdaemon.conf` file in a different place so that Ubuntu containers can still have it.  Maybe in `/usr/include/perfsonar/` ?  Other suggestions?
